protocols/imap/src/main/java/org/apache/james/imap/decode/parser/GetAnnotationCommandParser.java

```diff
@@ -0,0 +1,147 @@
+        super(ImapCommand.authenticatedStateCommand(ImapConstants.GETANNOTATION_COMMAND_NAME));
+    }
+
+    @Override
+    protected ImapMessage decode(ImapCommand command, ImapRequestLineReader request, String tag, ImapSession session)
+            throws DecodingException {
+        try {
+            GetAnnotationRequest annotationRequest = new GetAnnotationRequest(tag, command, request.mailbox());
+
+            while (nextNonSpaceChar(request) == '(') {
+                consumeParenthesis(request, annotationRequest);
+            }
+
+            if (ENDOFLINE_PATTERN.matches(nextNonSpaceChar(request))) {
+                consumeKey(request, annotationRequest);
+            }
+
+            Preconditions.checkArgument(annotationRequest.isValidParams());
+            return annotationRequest;
+        } catch (NullPointerException e) {
+            throw new DecodingException(HumanReadableText.ILLEGAL_ARGUMENTS, e.getMessage());
+        } catch (IllegalArgumentException e) {
+            throw new DecodingException(HumanReadableText.ILLEGAL_ARGUMENTS, e.getMessage());
+        }
+    }
+
+    private void consumeParenthesis(ImapRequestLineReader request, GetAnnotationRequest getAnnotationRequest) throws DecodingException {
+        request.consumeChar('(');
+        switch (request.nextChar()) {
+            case 'M':
+                getAnnotationRequest.setMaxsizeValue(consumeMaxsizeOpt(request));
+                break;
+
+            case 'D':
+                getAnnotationRequest.setDepthValue(consumeDepthOpt(request));
+                break;
+
+            default:
+                getAnnotationRequest.setKeys(consumeKeys(request));
+                break;
+        }
+        request.consumeChar(')');
+        if (!getAnnotationRequest.getKeys().isEmpty()) {
+            request.eol();
+        }
+    }
+
+    private Depth consumeDepthOpt(ImapRequestLineReader request) throws DecodingException {
+        request.consumeWord(new CharacterValidator() {
+            int pos = 0;
+            public boolean isValid(char chr) {
+                if (pos >= DEPTH.length) {
+                    return false;
+                } else {
+                    return DEPTH[pos++] == ImapRequestLineReader.cap(chr);
+                }
+            }
+        });
+        return Depth.fromString(request.atom());
+    }
+    public static void main(String[] agrs) {
+        System.out.println(Depth.fromString("-1"));
+    }
+
+    private long consumeMaxsizeOpt(ImapRequestLineReader request) throws DecodingException {
+        request.consumeWord(new CharacterValidator() {
+            int pos = 0;
+            public boolean isValid(char chr) {
+                if (pos >= MAXSIZE.length) {
+                    return false;
+                } else {
+                    return MAXSIZE[pos++] == ImapRequestLineReader.cap(chr);
+                }
+            }
+        });
+        return request.number(true);
```

Comment:
Why true ?

Why not simply number()  ?

When you have parameters like this, you should extract it to variables or constant to give them a name and make your code readable, because like this, I have to investigate to no what it means.